service.go

```diff
@@ -874,6 +879,37 @@ func handleServiceSASLReset(ctx *serviceContext, params []string) error {
 	return nil
 }
 
+func handleUserStatus(ctx *serviceContext, params []string) error {
+	// Limit to a small amount of users to avoid sending
+	// thousands of messages on large instances.
+	users := make([]*user, 0, 50)
+
+	ctx.user.srv.lock.Lock()
+	for _, user := range ctx.user.srv.users {
+		if len(users) == cap(users) {
+			break
+		}
+		users = append(users, user)
+	}
+	ctx.user.srv.lock.Unlock()
+
+	for _, user := range users {
+		line := user.Username
+		if user.Admin {
```

Comment:
Hm. This is racy: the user goroutine may mutate `*user.User`.

Can we switch the slice to `[]database.User` instead? That way, we'd copy the fields, and wouldn't be subject to race conditions.